src/test/java/fr/esgi/simple_auth_java/register/RegisterDBTest.java

```diff
@@ -0,0 +1,82 @@
+package fr.esgi.simple_auth_java.register;
+
+import fr.esgi.simple_auth_java.User;
+import fr.esgi.simple_auth_java.password_encrypt.PasswordEncryptSHA256;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.util.Scanner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * Created by Nicol on 10/06/2017.
+ */
+public class RegisterDBTest {
+    @Test
+    public void signUp() throws Exception {
+        //Init variables.
+        String email = "testsignup@esgitu.com";
+        String firstName = "Sign";
+        String lastName = "Up";
+        String password = "mysuperpasswd";
+
+        //Change them into 1 line configuration.
+        String simulatedUserInput = email + System.getProperty("line.separator") +
+                                    firstName + System.getProperty("line.separator") +
+                                    lastName + System.getProperty("line.separator") +
+                                    password + System.getProperty("line.separator");
+
+        //Convert it to a ByteArray of the System.In
+        ByteArrayInputStream setIn = new ByteArrayInputStream(simulatedUserInput.getBytes());
+
+        //Set it into System.In and called when System.In.next() fire.
+        System.setIn(setIn);
```

Comment:
bah à partir de cette ligne, `System.in` pointe sur `ByteArrayInputStream` même après la fin de ton test, car `System` est une variable globale de la JVM
donc tout le reste du programme va être affecté si tu ne remet pas l'input d'origine après ton test (qui pointe généralement sur la console)